todo_app/app.py

```diff
@@ -1,70 +1,51 @@
+api_token = os.getenv('API_TOKEN')
+item = Item(api_key, api_token)
+
 # Update the function that handles the GET request for fetching todo items
 @app.route('/')
 def index():
-    # Get Board ID, API key and API token from env file
+    # Get Board ID and List ID from env file
     list_id = os.getenv('TO_DO_LIST_ID')
-    api_key = os.getenv('API_KEY')
-    api_token = os.getenv('API_TOKEN')
-    # Use the fetch_todo_items function from trello_items module
-    todo_items = fetch_todo_items(list_id, api_key, api_token)
+    # Use the Item class method to fetch todo items
+    todo_items = item.fetch_todo_items(list_id)
     
     return render_template('index.html', items=todo_items)
 
-
-
-@app.route('/add', methods=['GET', 'POST'])
+@app.route('/add', methods=['POST'])
 def add_new_card():
-    # Get List ID, API key and API token from env file
+    # Get List ID from env file
     list_id = os.getenv('TO_DO_LIST_ID')
-    api_key = os.getenv('API_KEY')
-    api_token = os.getenv('API_TOKEN')
     card_name = request.form.get('item')
     updated_items = add_item(card_name)
-    to_do_list_items = fetch_todo_items(list_id, api_key, api_token)        
-  # Call the function to create a new card
-    card_id = create_todo_card(list_id, card_name, api_key, api_token)
+    to_do_list_items = item.fetch_todo_items(list_id)        
+    # Call the Item class method to create a new card
+    card_id = item.create_todo_card(list_id, card_name)
     
     return redirect('/')
 
-
-@app.route('/update', methods=['GET', 'POST'])
-def updates_card_to_done():
-    # Get Board ID, API key and API token from env file
-    to_do_list_id = os.getenv('TO_DO_LIST_ID') 
+@app.route('/update', methods=['POST'])
+def update_card_to_done():
+    # Get List IDs from env file
+    to_do_list_id = os.getenv('TO_DO_LIST_ID')
     done_list_id = os.getenv('DONE_LIST_ID')
-    #card_id = os.getenv('CARD_ID')
-    #board_id = os.getenv('BOARD_ID')
-    api_key = os.getenv('API_KEY')
-    api_token = os.getenv('API_TOKEN')
-    #name = "Done"
-    card_name_to_update = request.form.get('card_name')
-    #done_list_id = find_list_id_by_name(board_id, name, api_key, api_token)
-    card_id = find_card_id_by_name(to_do_list_id, card_name_to_update, api_key, api_token)
-    print (card_id)
-            
-  # Call function to change list ID of card from To Do to Done
-    update_new_card = update_card_to_done(done_list_id, card_id, api_key, api_token)
-    
-    #return redirect('/')
-    to_do_list_items = fetch_todo_items(to_do_list_id, api_key, api_token)
-    print (to_do_list_items)
-    #return render_template('index.html', items=to_do_list_items)
-    #return redirect('/')
-  
+    card_name_to_update = request.form.get('item')
+    # Use the Item class methods to find card ID and update the card
+    card_id = item.find_card_id_by_name(to_do_list_id, card_name_to_update)
+    update_new_card = item.update_card_to_done(done_list_id, card_id)
```

Comment:
Nice - this works well but it might give you a headache if users ever want to use the same to-do twice! ("I walked the dog yesterday, but I need to walk it again tomorrow!)

Generally we want to try use the ids as much as we can to keep that uniqueness, so we might for example:
* Move the "update" form in your HTML inside the for loop, so that we have a separate form for each to-do individually
* Change the `update_item_name` input to have a default value of `{{item.id}}` (which assumes the change above about using an Item class!)
* Hide that input, so that users don't have to know it exists (add the `hidden` keyword to the HTML element)
That way we could read out the item id rather than the name, and both speed up this function (by cutting out an API call to Trello) _and_ make it more reliable going forward

Reply:
Done.